grakn-graph/src/main/java/ai/grakn/graph/internal/ElementFactory.java

```diff
@@ -229,4 +230,10 @@ TypeImpl buildSpecificType(Vertex vertex, Type type){
     public EdgeImpl buildEdge(org.apache.tinkerpop.gremlin.structure.Edge edge, AbstractGraknGraph graknGraph){
         return new EdgeImpl(edge, graknGraph);
     }
+
+    private <X extends ConceptImpl> X buildConcept(Supplier<X> builder){
```

Comment:
As much as I like lambdas, I think you can just pass an `X` to this function instead!